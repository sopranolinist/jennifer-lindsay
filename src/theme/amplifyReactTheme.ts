import theme from './materialUi';

export default {
  navRight: { fontFamily: theme.typography.fontFamily },
  navButton: {
    backgroundColor: theme.palette.primary.main,
    fontFamily: theme.typography.fontFamily,
  },
  signInButton: {
    backgroundColor: theme.palette.primary.main,
    fontFamily: theme.typography.fontFamily,
  },
  button: {
    backgroundColor: theme.palette.primary.main,
    fontFamily: theme.typography.fontFamily,
  },
  a: {
    color: theme.palette.primary.main,
  },
  oAuthSignInButton: {
    color: 'white',
    backgroundColor: theme.palette.primary.main,
    fontFamily: theme.typography.fontFamily,
  },
  strike: {
    display: 'none',
  },
  googleSignInButton: {
    display: 'none',
  },
};
