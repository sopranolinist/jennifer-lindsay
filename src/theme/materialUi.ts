import { createTheme, ThemeOptions } from '@material-ui/core/styles';

function createMyTheme(options: ThemeOptions) {
  return createTheme({ ...options });
}

export const palette = {
  primary: {
    main: '#2e79b7',
    50: '#e6eff6',
    100: '#c0d7e9',
    200: '#97bcdb',
    300: '#6da1cd',
    400: '#4d8dc2',
    500: '#2e79b7',
    600: '#2971b0',
    700: '#2366a7',
    800: '#1d5c9f',
    900: '#124990',
    A100: '#c3dbff',
    A200: '#90bcff',
    A400: '#5d9dff',
    A700: '#438eff',
    contrastDefaultColor: 'light',
  },
  status: {
    approved: 'rgba(0, 255, 0, 0.5)',
    cancelled: 'rgba(192, 192, 192, 0.5)',
    denied: 'rgba(255,0,0,0.5)',
    pending: 'rgba(255, 255, 0, 0.5)',
  }
};
export const typography = {
  htmlFontSize: 12,
  fontFamily: '"Optima", "Roboto", sans-serif'
};

const MuiButtonBase = {
  root: {
    outline: '',
    WebkitTapHighlightColor: '',
    '-moz-appearance': '',
    '-webkit-appearance': '',
    '&::-moz-focus-inner': {
      borderStyle: ''
    }
  }
};

export default createMyTheme({
  typography,
  palette,
  overrides: {
    MuiButtonBase // WCAG 2 compliance, enables visual focus state
  }
});
