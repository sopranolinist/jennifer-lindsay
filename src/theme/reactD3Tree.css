/* overriding react-d3-tree styles that can't be overriden from the library */

.leafNodeBase {
  cursor: inherit;
}
