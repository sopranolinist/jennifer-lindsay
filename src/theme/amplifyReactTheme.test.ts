import theme from './amplifyReactTheme';

describe('Theme - Amplify React', () => {
  it('exports and object', () => {
    expect(theme).toMatchSnapshot();
  });
});
