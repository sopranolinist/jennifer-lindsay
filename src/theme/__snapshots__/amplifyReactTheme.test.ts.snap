// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme - Amplify React exports and object 1`] = `
Object {
  "a": Object {
    "color": "#2e79b7",
  },
  "button": Object {
    "backgroundColor": "#2e79b7",
    "fontFamily": "\\"Roboto\\", sans-serif",
  },
  "googleSignInButton": Object {
    "display": "none",
  },
  "navButton": Object {
    "backgroundColor": "#2e79b7",
    "fontFamily": "\\"Roboto\\", sans-serif",
  },
  "navRight": Object {
    "fontFamily": "\\"Roboto\\", sans-serif",
  },
  "oAuthSignInButton": Object {
    "backgroundColor": "#2e79b7",
    "color": "white",
    "fontFamily": "\\"Roboto\\", sans-serif",
  },
  "signInButton": Object {
    "backgroundColor": "#2e79b7",
    "fontFamily": "\\"Roboto\\", sans-serif",
  },
  "strike": Object {
    "display": "none",
  },
}
`;
